src/app/services/xml-parsers/basic-parsers.ts

```diff
@@ -0,0 +1,131 @@
+import { AttributesMap } from 'ng-dynamic-component';
+import { Attributes, GenericElement, Lb, Note, NoteLayout, Paragraph, Text, XMLElement } from '../../models/evt-models';
+import { isNestedInElem, xpath } from '../../utils/dom-utils';
+import { replaceMultispaces } from '../../utils/xml-utils';
+import { createParser, getClass, getID, parseChildren, ParseFn, Parser } from './parser-models';
+
+export class EmptyParser {
+    genericParse: ParseFn;
+    constructor(parseFn: ParseFn) { this.genericParse = parseFn; }
+}
+
+export class AttributeParser extends EmptyParser implements Parser<XMLElement> {
+    parse(data: HTMLElement): Attributes {
+        return Array.from(data.attributes)
+            .map(({ name, value }) => ({ [name === 'xml:id' ? 'id' : name.replace(':', '-')]: value }))
+            .reduce((x, y) => ({ ...x, ...y }), {});
+    }
+}
+export class AttributeMapParser extends EmptyParser implements Parser<XMLElement> {
+    parse(xml: XMLElement) {
+        const attributes: AttributesMap = {};
+        Array.from(xml.attributes).forEach((attr) => {
+            attributes[attr.name] = attr.value;
+        });
+
+        return attributes;
+    }
+}
+
+export class TextParser implements Parser<XMLElement> {
+    parse(xml: XMLElement): Text {
+        return {
+            type: Text,
+            text: replaceMultispaces(xml.textContent),
+            attributes: {},
+        } as Text;
+    }
+}
+
+export class ParagraphParser extends EmptyParser implements Parser<XMLElement> {
+    attributeParser = createParser(AttributeParser, this.genericParse);
+    parse(xml: XMLElement): Paragraph {
+        const attributes = this.attributeParser.parse(xml);
+        const paragraphComponent: Paragraph = {
+            type: Paragraph,
+            content: parseChildren(xml, this.genericParse),
+            attributes,
+            n: attributes.n ?? '-1', // TODO: Check it this can become a number
```

Comment:
Remove `'-1'`: if no `@n` attribute there should be no `n` value for paragraph. It can't become a number since in TEI its value is a `teidata.text` and can also accept a non-numeric label.